### ISSUE TITLE
When blocky is used as the upstream of other dns forwarders

### ISSUE THREAD

**Issue by liang-hiwin** [0xERR0R/blocky#952]
When blocky is used as the upstream of other dns forwarders, client_ip and client_names are not displayed as the real ip of the client, but are displayed as the internal network 127.0.0.1 ip

**Comment by liang-hiwin** [comment]
[2023-03-26 00:59:45] DEBUG upstream_resolver: received response from upstream answer=A (117.148.167.25), A (117.148.167.18),  A (117.148.167.26) **/** net=https protocol=UDP question=A (psbi.cname.getui.com.) response_time_ms=3 return_code=NOERROR upstream=https://google.com/dns-query upstream_ip=8.8.8.8
```

```

**Comment by kwitsch** [comment]
If blocky is used as upstream the other DNS forwarder is the client.
In your case I guess you're running both on the same machine?
Therefore the IP 127.0.0.1 is correct.

It could possibly be implemented using [RFC7871](https://datatracker.ietf.org/doc/html/rfc7871). This would require the forwarding resolver which received the original request to also implement it.

**Comment by liang-hiwin** [comment]
> If blocky is used as upstream the other DNS forwarder is the client. In your case I guess you're running both on the same machine? Therefore the IP 127.0.0.1 is correct.
> 
> It could possibly be implemented using [RFC7871](https://datatracker.ietf.org/doc/html/rfc7871). This would require the forwarding resolver which received the original request to also implement it.

Yes, I'm running on the same server, it's fine if it doesn't affect the ecs client subnet requests.

**Comment by liang-hiwin** [comment]
I still have a question, what is the port of Prometheus, the document is not written

**Comment by kwitsch** [comment]
Can this issue be closed or should it be converted to a feature request for RFC7871 to be implemented?

**Comment by liang-hiwin** [comment]
> [RFC7871](https://datatracker.ietf.org/doc/html/rfc7871). This would require the forwarding resolver which received the original request to also implement it.

Use RFC7871 to achieve
